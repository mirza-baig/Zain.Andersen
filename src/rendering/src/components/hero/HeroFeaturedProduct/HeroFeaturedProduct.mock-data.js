const defaultData = {
  rendering: {
    dataSource: 'sampledatasource',
    componentName: 'HeroFeaturedProduct',
  },
  fields: {
    subheading1: {
      value:
        '<h3>Example subheading with <a class="text-primary font-bold cursor-pointer">link</a> goes here. </h3>',
    },
    subheading2: {
      value: '<h3>Example subheading  goes here. </h3>',
    },
    subheading3: {
      value: '<h3>Example subheading  goes here. </h3>',
    },
    subheading4: {
      value: '<h3>Example subheading  goes here. </h3>',
    },
    subheading5: {
      value: '<h3>Example subheading  goes here. </h3>',
    },
    componentSpacing: null,
    sectionId: {
      value: '',
    },
    theme: {
      value: '',
    },
    subheadlineText: {
      value: 'Small Heading',
    },
    headlineText: {
      value: 'Large Heading',
    },
    primaryImage: {
      value: {
        src: 'https://i.postimg.cc/1zcs4Xcv/AW-400-S-Window-Door-Hero-400height-3window.png',
        alt: '',
        width: '574',
        height: '574',
      },
    },
    primaryImageMobile: {
      value: {
        src: 'https://i.postimg.cc/1zcs4Xcv/AW-400-S-Window-Door-Hero-400height-3window.png',
        alt: '',
        width: '574',
        height: '574',
      },
    },
    primaryImageMobileFocusArea: {
      id: 'e6b36945-ebc0-4af8-ad95-d8268b56b7d0',
      url: 'http://localhost/sitecore/login/sitecore/system/Settings/Feature/EnterpriseWeb/Enums/Image-Focus/Center',
      name: 'Center',
      displayName: 'Center',
      fields: {
        Value: {
          value: 'center',
        },
      },
      templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
      templateName: 'Enum',
    },
    eventName: {
      value: '',
    },
    eventZone: {
      value: '',
    },
    eventType: {
      value: '',
    },
  },
};

export default defaultData;
