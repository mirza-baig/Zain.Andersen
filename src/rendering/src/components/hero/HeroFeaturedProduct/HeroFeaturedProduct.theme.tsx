// Lib
import { ThemeFile } from 'lib/context/ThemeContext';

export const HeroFeaturedProductTheme: ThemeFile = {
  aw: {
    classes: {
      productWrapper: 'relative bg-light-gray text-center pt-xs md:pt-l',
      headingsWrapper: 'font-bold',
      smallHeadline: 'text-m text-dark-gray',
      largeHeadline: 'text-sm-xxl md:text-xxl -mb-s',
      imageWrapper: '-mb-s ml:-mb-xl h-full',
      subheadingsList:
        'flex flex-col md:flex-row justify-center text-center mt-s md:mt-l !font-sans',
      subheadingItem:
        'relative px-m md:px-l py-s first:before:hidden before:absolute before:content-[""] before:w-full md:before:w-[1px] before:h-[1px] md:before:h-full before:bg-secondary before:top-0 before:left-0',
      rteClasses: 'text-sm-s font-heavy md:text-s [&_a]:font-sans',
      additionalDesktopClasses: 'relative h-[310px] md:h-[592px] md:max-w-[592px] mx-auto',
      additionalMobileClasses: 'mx-auto',
    },
  },
  rba: {
    classes: {
      productWrapper: 'relative pt-l md:pt-xl',
      headingsWrapper: 'font-bold md:font-medium ml:absolute ml:top-1/2 ml:-translate-y-1/2 w-1/2',
      smallHeadline: 'text-xxs md:text-s font-bold md:font-medium !font-serif md:!font-sans',
      largeHeadline: 'text-l md:text-xxl font-medium',
      imageWrapper: 'mt-s -mb-l w-full',
      subheadingsList: 'flex flex-col md:flex-row mt-s ml:mt-l ml:pt-s !font-sans',
      subheadingItem:
        'flex-1 relative py-xs before:absolute before:content-[""] before:w-m md:before:w-l before:h-[3px] before:bg-primary before:top-0 before:left-0',
      rteClasses:
        '[&_p]:font-serif md:[&_p]:!font-sans md:[&_a]:!font-sans text-xs md:text-m font-bold md:font-medium',
      additionalDesktopClasses: 'relative h-full w-full mx-auto ml:float-right -mb-s ml:-mb-xl',
      additionalMobileClasses: 'relative mx-auto',
    },
  },
};
