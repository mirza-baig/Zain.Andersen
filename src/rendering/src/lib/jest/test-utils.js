// Global
import { render, fireEvent, waitFor } from '@testing-library/react';
import { ComponentPropsContext } from '@sitecore-jss/sitecore-jss-nextjs';
import { FastSitecoreContext } from 'lib/overrides/FastSitecoreContext';

// Lib
import { createSampleComponentFactory } from 'lib/mocks/mock-placeholder';

export const eeSitecoreContext = {
  route: {
    name: 'Foo',
    placeholders: {},
  },
  context: {
    pageEditing: true,
    site: {
      name: 'JssTestWeb',
    },
    language: 'en',
  },
};

export const renderComponent = (
  Component,
  {
    componentProps = {},
    staticProps = {},
    componentFactory = createSampleComponentFactory(),
    sitecoreContext = {
      route: {
        name: 'Foo',
        placeholders: {},
      },
    },
  }
) => {
  const mockLayoutData = {
    sitecore: sitecoreContext,
  };

  return render(
    <ComponentPropsContext value={staticProps}>
      <FastSitecoreContext componentFactory={componentFactory} layoutData={mockLayoutData}>
        <Component {...componentProps} />
      </FastSitecoreContext>
    </ComponentPropsContext>
  );
};

export const snapshot = (Component, mockData) => {
  const component = renderComponent(Component, mockData);
  expect(component.container.firstChild).toMatchSnapshot();
  return component;
};

export const hasDataComponent = async (component, name) => {
  await waitFor(() => {
    const nodes = component.baseElement.querySelectorAll(`[data-component="${name}"]`);
    return expect(nodes.length).toBe(1);
  });
};

export const mockDateAs = (epochMs) => {
  if (typeof epochMs !== 'number') {
    console.error(
      'mockDateAs only accepts epoch values in milliseconds to prevent timezones from factoring in to Date creation.'
    );
  }

  jest.spyOn(global.Date, 'now').mockImplementation(() => new Date(epochMs).valueOf());
};

export const testFieldError = async (Component, mockData, { label, value, error }) => {
  const component = renderComponent(Component, mockData);
  const field = component.getByLabelText(label);
  fireEvent.change(field, { target: { value } });
  field.blur();
  await waitFor(() => {
    const err = component.queryByText(error);
    return expect(err).not.toBe(null);
  });
};
