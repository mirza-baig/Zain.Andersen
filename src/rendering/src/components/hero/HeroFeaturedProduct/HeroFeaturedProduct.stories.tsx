// Global
import { Story, Meta } from '@storybook/react';
// Local
import HeroFeaturedProduct, { HeroFeaturedProductProps } from './HeroFeaturedProduct';
import defaultData from './HeroFeaturedProduct.mock-data';

export default {
  title: 'Hero/HeroFeaturedProduct',
  component: HeroFeaturedProduct,
} as Meta;
const Template: Story<HeroFeaturedProductProps> = (props) => <HeroFeaturedProduct {...props} />;
export const Default = Template.bind({});
Default.args = defaultData;
