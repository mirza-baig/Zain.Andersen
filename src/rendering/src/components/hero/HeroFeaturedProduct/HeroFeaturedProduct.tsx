// Global
import { Feature } from 'src/.generated/Feature.EnterpriseWeb.model';
import { useTheme } from 'lib/context/ThemeContext';
import { Field, withDatasourceCheck } from '@sitecore-jss/sitecore-jss-nextjs';
// Components
import classNames from 'classnames';
import { Component } from 'src/helpers/Component';
import { Subheadline } from 'src/helpers/Subheadline';
import { Headline } from 'src/helpers/Headline';
import { ImagePrimary } from 'src/helpers/Media';
import { RichTextWrapper } from 'src/helpers/RichTextWrapper';
import { HeroFeaturedProductTheme } from './HeroFeaturedProduct.theme';

export type HeroFeaturedProductProps =
  Feature.EnterpriseWeb.Enterprise.Components.Hero.HeroFeaturedProduct;
const HeroFeaturedProduct = (props: HeroFeaturedProductProps) => {
  const { fields } = props;

  const { themeName, themeData } = useTheme(HeroFeaturedProductTheme);
  const subheadingList: Array<Field<string>> = [];

  if (!fields) {
    return null;
  }

  Object.keys(fields).map((key, index) => {
    if (key === `subheading${index + 1}`) {
      fields[key].value && subheadingList.push(fields[key]);
    }
  });

  return (
    <>
      <Component
        variant="lg"
        sectionWrapperClasses={themeName === 'rba' ? 'bg-light-gray' : undefined}
        dataComponent="hero/herofeaturedproduct"
        {...props}
      >
        <div className={classNames('col-span-12', themeData.classes.productWrapper)}>
          <div className={themeData.classes.headingsWrapper}>
            <Subheadline classes={themeData.classes.smallHeadline} useTag="h2" {...props} />
            <Headline classes={themeData.classes.largeHeadline} useTag="h1" {...props} />
          </div>
          <div className={themeData.classes.imageWrapper}>
            <ImagePrimary
              maxH={'max-w-[592px]'}
              additionalDesktopClasses={themeData.classes.additionalDesktopClasses}
              additionalMobileClasses={themeData.classes.additionalDesktopClasses}
              {...props}
              priority
            />
          </div>
        </div>
      </Component>
      <Component variant="lg" {...props}>
        <div className={`col-span-12 ${themeData.classes.subheadingsList}`}>
          {subheadingList.map((item, index: number) => {
            return (
              <div key={index} className={classNames(themeData.classes.subheadingItem)}>
                <RichTextWrapper field={item} classes={themeData.classes.rteClasses} />
              </div>
            );
          })}
        </div>
      </Component>
    </>
  );
};

export default withDatasourceCheck()<HeroFeaturedProductProps>(HeroFeaturedProduct);
