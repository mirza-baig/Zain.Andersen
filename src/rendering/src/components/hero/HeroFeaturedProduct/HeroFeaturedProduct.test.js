// Global
import { hasDataComponent, snapshot } from 'lib/jest/test-utils';
// Local
import HeroFeaturedProduct from './HeroFeaturedProduct';
import defaultData from './HeroFeaturedProduct.mock-data';

it('renders correctly', async () => {
  const component = snapshot(HeroFeaturedProduct, { componentProps: defaultData });
  await hasDataComponent(component, 'hero/herofeaturedproduct');
});
