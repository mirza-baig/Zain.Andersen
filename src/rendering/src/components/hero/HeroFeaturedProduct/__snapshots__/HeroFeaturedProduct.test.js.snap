// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  data-component="hero/herofeaturedproduct"
  id="id1695242607"
>
  <div
    class="my-8 lg:mx-auto md:max-w-screen-lg theme-white"
  >
    <div
      class="section-grid grid grid-cols-2 md:grid-cols-12 px-m gap-s md:gap-s "
    >
      <div
        class="col-span-12 relative bg-light-gray text-center pt-xs md:pt-l"
      >
        <div
          class="font-bold"
        >
          <div
            class="font-sans text-m text-dark-gray"
          >
            <h2>
              Small Heading
            </h2>
          </div>
          <div
            class="items-top font-sans [&_a:hover]:underline text-sm-xxl md:text-xxl -mb-s"
          >
            <h1>
              Large Heading
            </h1>
          </div>
        </div>
        <div
          class="-mb-s ml:-mb-xl h-full"
        >
          <div
            class="relative hidden md:block aspect-picture max-w-[592px]   relative h-[310px] md:h-[592px] md:max-w-[592px] mx-auto"
          >
            <span
              style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 100% 0px 0px 0px;"
              />
              <img
                alt=""
                data-nimg="responsive"
                decoding="async"
                sizes="100vw"
                src="/_next/image/?url=https%3A%2F%2Fi.postimg.cc%2F1zcs4Xcv%2FAW-400-S-Window-Door-Hero-400height-3window.png&w=3840&q=75"
                srcset="/_next/image/?url=https%3A%2F%2Fi.postimg.cc%2F1zcs4Xcv%2FAW-400-S-Window-Door-Hero-400height-3window.png&w=640&q=75 640w, /_next/image/?url=https%3A%2F%2Fi.postimg.cc%2F1zcs4Xcv%2FAW-400-S-Window-Door-Hero-400height-3window.png&w=750&q=75 750w, /_next/image/?url=https%3A%2F%2Fi.postimg.cc%2F1zcs4Xcv%2FAW-400-S-Window-Door-Hero-400height-3window.png&w=828&q=75 828w, /_next/image/?url=https%3A%2F%2Fi.postimg.cc%2F1zcs4Xcv%2FAW-400-S-Window-Door-Hero-400height-3window.png&w=1080&q=75 1080w, /_next/image/?url=https%3A%2F%2Fi.postimg.cc%2F1zcs4Xcv%2FAW-400-S-Window-Door-Hero-400height-3window.png&w=1200&q=75 1200w, /_next/image/?url=https%3A%2F%2Fi.postimg.cc%2F1zcs4Xcv%2FAW-400-S-Window-Door-Hero-400height-3window.png&w=1920&q=75 1920w, /_next/image/?url=https%3A%2F%2Fi.postimg.cc%2F1zcs4Xcv%2FAW-400-S-Window-Door-Hero-400height-3window.png&w=2048&q=75 2048w, /_next/image/?url=https%3A%2F%2Fi.postimg.cc%2F1zcs4Xcv%2FAW-400-S-Window-Door-Hero-400height-3window.png&w=3840&q=75 3840w"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
            </span>
          </div>
          <div
            class="relative block w-full md:hidden relative h-[310px] md:h-[592px] md:max-w-[592px] mx-auto"
          >
            <span
              style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 100% 0px 0px 0px;"
              />
              <img
                alt=""
                data-nimg="responsive"
                decoding="async"
                sizes="100vw"
                src="/_next/image/?url=https%3A%2F%2Fi.postimg.cc%2F1zcs4Xcv%2FAW-400-S-Window-Door-Hero-400height-3window.png&w=3840&q=75"
                srcset="/_next/image/?url=https%3A%2F%2Fi.postimg.cc%2F1zcs4Xcv%2FAW-400-S-Window-Door-Hero-400height-3window.png&w=640&q=75 640w, /_next/image/?url=https%3A%2F%2Fi.postimg.cc%2F1zcs4Xcv%2FAW-400-S-Window-Door-Hero-400height-3window.png&w=750&q=75 750w, /_next/image/?url=https%3A%2F%2Fi.postimg.cc%2F1zcs4Xcv%2FAW-400-S-Window-Door-Hero-400height-3window.png&w=828&q=75 828w, /_next/image/?url=https%3A%2F%2Fi.postimg.cc%2F1zcs4Xcv%2FAW-400-S-Window-Door-Hero-400height-3window.png&w=1080&q=75 1080w, /_next/image/?url=https%3A%2F%2Fi.postimg.cc%2F1zcs4Xcv%2FAW-400-S-Window-Door-Hero-400height-3window.png&w=1200&q=75 1200w, /_next/image/?url=https%3A%2F%2Fi.postimg.cc%2F1zcs4Xcv%2FAW-400-S-Window-Door-Hero-400height-3window.png&w=1920&q=75 1920w, /_next/image/?url=https%3A%2F%2Fi.postimg.cc%2F1zcs4Xcv%2FAW-400-S-Window-Door-Hero-400height-3window.png&w=2048&q=75 2048w, /_next/image/?url=https%3A%2F%2Fi.postimg.cc%2F1zcs4Xcv%2FAW-400-S-Window-Door-Hero-400height-3window.png&w=3840&q=75 3840w"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain; object-position: center;"
              />
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;
